addons/account/models/account_move.py

```diff
@@ -288,6 +288,13 @@ def _get_default_invoice_incoterm(self):
     inalterable_hash = fields.Char(string="Inalterability Hash", readonly=True, copy=False)
     string_to_hash = fields.Char(compute='_compute_string_to_hash', readonly=True)
 
+    _sql_constraints = [
+        ('unique_name',
+         """EXCLUDE (name WITH =, journal_id WITH =, move_type WITH =)
+              WHERE (state = 'posted' AND name != '/')""",
```

Comment:
I guess you cannot use a UNIQUE constraint or a UNIQUE INDEX because of the WHERE clause.  Is that right?

Reply:
`UNIQUE INDEX` works, but there was no easy way to have a nice error message. I tried by adding a2575512e45aace6c36f338ef20dc7f96607b249 and 86a7f92 but others here preferred the exclusion constraint.

Reply:
A good error message is a good reason for the constraint!

Reply:
I had a good error message thanks to 86a7f92